code/game/machinery/machinery.dm

```diff
@@ -917,3 +917,30 @@ Class Procs:
 				table_shift()
 		else
 			table_unshift()
+
+/obj/machinery/proc/cardinalize_dir()
+ //bit operation that ensures cardinal facing in the off chance its not.
+	dir = dir^(dir&(dir-1))
+
+/obj/machinery/proc/getDirString(var/atom/target)
```

Comment:
wouldn't it make more sense to make these two global procs?

Reply:
Good call